rfcs/000-consolidate-amundsen-microservices.md

```diff
@@ -0,0 +1,154 @@
+We will add a deprecation warning on each readme and packages page where applicable, and introduce the new way i.e., a mono-repo, along the way. All the new work will be done in the new repository. 
+
+## Drawbacks
+
+For each phase mentioned above, we may need to freeze development to the respective repository. Although it should not take too long to merge multiple repositories, there still can be unforeseen issues that delay the process.
+
+## Alternatives
+
+1. We do not change anything and live with the existing architecture. This will eventually result in dependency hell, redundant code, and a complex network of repositories.
+
+1. An alternative could be to merge all 4 microservices into a single repository i.e, frontend, metadata, search, and common would simply become `amundsen-io/amundsen` repository. 
+
+    For proxies we'll have separate packages. i.e., `amundsen-io/metadataproxy` and `amundsen-io/searchproxy`. 
+    - metaproxy: Will hold the neo4j, atlas, rdbms, neptune etc. 
+    - searchproxy: will have the implementation of elasticsearch, atlas etc., 
+
+    We'll install amundsen as `pip install amundsen`, and this will install the amundsen project without any proxy for metadata or search, hence make it a core project without any custom dependencies for anyone. 
+
+    Config option in amundsen repository, to select the proxy will be: 
+
+    - METADATA_PROXY = "amundsen_metadataproxy.proxy.neo4j" 
+    - SEARCH_PROXY = "amundsen_searchproxy.proxy.elasticsearch" 
+
+    and we'll install proxies like this. 
+    
+    - `pip install amundse-metadataproxy[neo4j]`
+    - `pip install amundsen-searchproxy[elasticsearch]` 
+
+    This will make the whole codebase much lighter and not dependent on any specific proxies or packages like neo4j, apache-atlas, neptune, etc., 
+
+1. Deprecate the split of metadata, search, and amundsencommon repositories, and merge them all into one mono repository, and call this `amundsenbackend`. Remove the Flask based code from the frontend repository and call the metadata/search endpoints directly via the React app. 
+    Summary: The split will instead be based on the backend and frontend.
+    Following will be the repositories (and packages) of Amundsen after the change.
+	- amundsenfrontend (Frontend React Application)
+	- amundsenbackend (frontend flask application, metadata, search, and common repositories consolidated in this repository)
+	- amundsendatabuilder (Same as today's databuilder repository)
+    
+
+## Unresolved questions
+TBD
+> What parts of this RFC are TBD?
+
+## Future possibilities
+
+This change can pave our ways to move Amundsen towards the next generation event-based system. More contributors and adopters will result in more integrations and diversity of features. 
+
+Ref: https://engineering.linkedin.com/blog/2020/datahub-popular-metadata-architectures-explained
```

Comment:
even you take the other project for example, I know at Li, the GMS is a separate deployment (https://github.com/linkedin/datahub-gma) which dh is only one of the app on top of the backend.